data-prepper-plugins/grok-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/grok/GrokPrepper.java

```diff
@@ -123,33 +115,52 @@ public GrokPrepper(final PluginSetting pluginSetting) {
             } catch (JsonProcessingException e) {
                 LOG.error("Failed to parse the record [{}]", record.getData());
                 recordsOut.add(record);
+            } catch (TimeoutException e) {
+                LOG.error("Matching on record [{}] took longer than [{}] and timed out", record.getData(), grokPrepperConfig.getTimeoutMillis());
+                recordsOut.add(record);
+            } catch (ExecutionException e) {
+                LOG.error("An exception occurred while matching on record [{}]", record.getData(), e);
+                recordsOut.add(record);
+            } catch (InterruptedException e) {
+                LOG.error("Matching on record [{}] was interrupted", record.getData(), e);
+                recordsOut.add(record);
+            } catch (RuntimeException e) {
+                LOG.error("Unknown exception occurred when matching record [{}]", record.getData(), e);
+                recordsOut.add(record);
             }
-        }
+         }
         return recordsOut;
     }
 
     @Override
     public void prepareForShutdown() {
-
+        executorService.shutdown();
     }
 
     @Override
     public boolean isReadyForShutdown() {
+        try {
+            if (executorService.awaitTermination(grokPrepperConfig.getTimeoutMillis(), TimeUnit.MILLISECONDS)) {
```

Comment:
Should we couple the grok prepper timeout with the executorService shutdown? How long does this shutdown typically take? Do we want users to be able to control this timeout?

Reply:
This is a valid point, especially given that the timeoutMillis can be 0, which would mess up this implementation. We don't want users to be able to control this timeout. I think an arbitrary value of 100 ms is fine here.